lib/services/plugins/npm-plugins-service-base.ts

```diff
@@ -0,0 +1,290 @@
+			let npmAuthorsGroup = "((?:=\\S+\\s?)+)\\s+";
+
+			// Should look like this /(\S+)\s+([^=]+)((?:=\S+\s?)+)\s+(\d+\-\d+\-\d+)\s(\S+)\s+([^=]+)/
+			let pluginRowRegExp = new RegExp(`${npmNameGroup}${npmFreeTextGroup}${npmAuthorsGroup}${npmDateGroup}${npmNameGroup}${npmFreeTextGroup}`);
+
+			_.each(pluginsRows, (pluginRow: string) => {
+				let matches = pluginRowRegExp.exec(pluginRow.trim());
+
+				if (!matches || !matches[0]) {
+					return;
+				}
+
+				pluginsFound.push({
+					name: matches[1],
+					description: matches[2],
+					version: matches[5]
+				});
+			});
+
+			return pluginsFound;
+		}).future<IBasicPluginInformation[]>()();
+	}
+
+	public fetch(pluginIdentifier: string): IFuture<void> {
+		return (() => {
+			this.$project.ensureProject();
+			if (!pluginIdentifier) {
+				this.$errors.fail("You must specify local path, URL to a plugin repository, name or keywords of a plugin published to the NPM.");
+			}
+
+			if (this.isUrlToRepository(pluginIdentifier) || this.isLocalPath(pluginIdentifier).wait()) {
+				if (this.hasTgzExtension(pluginIdentifier)) {
+					pluginIdentifier = path.resolve(pluginIdentifier);
+				}
+
+				this.fetchPluginCore(pluginIdentifier).wait();
+				return;
+			}
+
+			let plugin = _.find(this.getAvailablePlugins(), (pl: IPlugin) => pl.data.Identifier.toLowerCase() === pluginIdentifier || pl.data.Name.toLowerCase() === pluginIdentifier);
+			let pluginUrl = plugin && plugin.data && plugin.data.Url ? plugin.data.Url : null;
+			let plugins = this.findPlugins([pluginIdentifier]).wait();
+			let pluginKeys = _.map(plugins, (pluginInfo: IBasicPluginInformation) => pluginInfo.name);
+			let pluginsCount = pluginKeys.length;
+
+			if (pluginsCount === 0) {
+				if (pluginUrl) {
+					try {
+						// The plugin is not in npm but it is in our marketplace.
+						this.fetchPluginCore(pluginUrl).wait();
+					} catch (error) {
+						this.$logger.info("The plugin cannot be downloaded using npm, because it has no package.json in it. You can still download it from this link: " + plugin.data.Url.grey);
+					}
+				} else {
+					this.fetchPluginCore(pluginIdentifier).wait();
+				}
+
+				return;
+			}
+
+			if (pluginsCount > 1 && pluginKeys[0] !== pluginIdentifier) {
+				if (commonHelpers.isInteractive()) {
+					let selectedPlugin = this.$prompter.promptForChoice("We found multiple plugins with your search parameters please choose which one you want to fetch.", pluginKeys).wait();
+					this.fetchPluginCore(selectedPlugin).wait();
+				} else {
+					this.$logger.out("There are more then 1 matching plugins: " + pluginKeys.join(", ") + ".");
+				}
+
+				return;
+			}
+
+			try {
+				this.fetchPluginCore(pluginKeys[0]).wait();
+			} catch (err) {
+				if (pluginUrl) {
+					this.$logger.trace("Error while trying to fetch plugin with id " + pluginIdentifier + " via npm. Error is: " + err.message + ".");
+					this.fetchPluginCore(pluginUrl).wait();
+				} else {
+					this.$errors.fail(err.message);
```

Comment:
`failWithoutHelp` maybe - user does not need our help in this case, as he knows how to use the command.